subways/structure/route.py

```diff
@@ -33,24 +33,27 @@
 DISALLOWED_ANGLE_BETWEEN_STOPS = 20  # in degrees
 
 
-def get_start_end_times(
+def parse_time_range(
     opening_hours: str,
-) -> tuple[tuple[int, int], tuple[int, int]] | tuple[None, None]:
+) -> tuple[tuple[int, int], tuple[int, int]] | None:
     """Very simplified method to parse OSM opening_hours tag.
     We simply take the first HH:MM-HH:MM substring which is the most probable
     opening hours interval for the most of the weekdays.
     """
     start_time, end_time = None, None
     m = START_END_TIMES_RE.match(opening_hours)
```

Comment:
I presume `opening_hours` can have any format specified in [wiki](https://wiki.openstreetmap.org/wiki/Key:opening_hours)

If all we need from this regular expression is time HH:MM-HH:MM, then we can use
`START_END_TIMES_RE = re.compile(r"\d{2}):(\d{2})-(\d{2}):(\d{2})"` https://regex101.com/r/EYy5ud/1

Reply:
Timetables may be too complex, we don't set a goal to process all of them. As stated in the doc-string, we take first interval-like string and consider it the most important time interval for the route. See the test case for the idea: https://github.com/alexey-zakharenkov/subways/blob/process-route-duration/subways/tests/test_route.py#L50

It's too far from the moment the applications that use this Validator to take into account complex timetables.